fili-core/src/main/java/com/yahoo/bard/webservice/web/ApiHaving.java

```diff
@@ -58,7 +58,11 @@
      *
      * @throws BadHavingException  when having pattern is not matched or when any of its properties are not valid.
      */
-    public ApiHaving(@NotNull String havingQuery, MetricDictionary metricDictionary) throws BadHavingException {
+    public ApiHaving(
+            @NotNull String havingQuery,
+            Map<String, LogicalMetric> metricDictionary
```

Comment:
can be unwrapped

Reply:
No we can't. The exception sends it over 120 characters.